driver-simulator/simulate.js

```diff
@@ -0,0 +1,55 @@
+const http = require("http");
+const vehicles = require("./vehicles.json");
+let stepIndexes = {};
+
+const API_HOST = process.argv[2].split(":")[0];
+const API_PORT = process.argv[2].split(":")[1];
+
+vehicles.forEach(vehicle => {
+  registerVehicle(vehicle.id);
+
+  setInterval(() => {
+    let stepIndex =
+      stepIndexes[vehicle.id] === undefined ? 0 : stepIndexes[vehicle.id] + 1;
+    stepIndexes[vehicle.id] = stepIndex;
+
+    if (stepIndex < vehicle.steps.length) {
+      updateVehicleLocation(vehicle.id, vehicle.steps[stepIndex]);
+    } else if (stepIndex === vehicle.steps.length) {
+      deregisterVehicle(vehicle.id);
+    }
+  }, 3000);
+});
+
+function updateVehicleLocation(vehicleId, location) {
+  request(
+    `/vehicles/${vehicleId}/locations`,
+    "POST",
+    { lat: location[1], lat: location[0], at: new Date().toISOString() },
```

Comment:
You have two `lat` values here.

It should be: `{ lat: location[0], lng: location[1], /* ... */ }`

Reply:
Good catch!